src/core/net/tcp6.hpp

```diff
@@ -0,0 +1,302 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions for UDP/IPv6 sockets.
+ */
+
+#ifndef TCP6_HPP_
+#define TCP6_HPP_
+
+#include "openthread-core-config.h"
+
+#include <openthread/tcp.h>
+
+#include "net/ip6_headers.hpp"
+#include "net/socket.hpp"
+
+#include "common/linked_list.hpp"
+#include "common/locator.hpp"
+#include "common/non_copyable.hpp"
+
+namespace ot {
+namespace Ip6 {
+
+class Udp;
+
+/**
+ * @addtogroup core-tcp
+ *
+ * @brief
+ *   This module includes definitions for TCP/IPv6 sockets.
+ *
+ * @{
+ *
+ */
+class Tcp : public InstanceLocator, private NonCopyable
+{
+public:
+    /**
+     * This class represents an endpoint of a TCP/IPv6 connection.
+     *
+     */
+    class Endpoint : public otTcpEndpoint, public LinkedListEntry<Endpoint>
+    {
+        friend class Tcp;
+        friend class LinkedList<Endpoint>;
+    };
+
+    /**
+     * This class represents a TCP/IPv6 listener.
```

Comment:
```suggestion
     * This class represents a TCP/IPv6 listener.
     *
```

Reply:
Thanks.

Reply:
I'll apply this convention uniformly throughout in the next update.